src/mo_frontend/test_field_srcs.ml

```diff
@@ -0,0 +1,460 @@
+  let* _typ, sscope =
+    Mo_types.Cons.session ~scope:filename (fun () ->
+      Mo_frontend.Typing.infer_prog
+        ~viper_mode:false
+        senv
+        None
+        async_cap
+        prog)
+  in
+  Diag.return (prog, sscope)
+
+let check_builtin what src senv0 : Mo_def.Syntax.prog * Mo_types.Scope.t =
+  let builtin_error phase what (msgs : Diag.messages) =
+    Printf.eprintf "%s %s failed\n" phase what;
+    Diag.print_messages msgs;
+    exit 1
+  in
+  match parse_from_string Mo_frontend.Lexer.mode_priv what src with
+  | Error es -> builtin_error "parsing" what es
+  | Ok (prog, _ws) ->
+    match infer_prog prog senv0 Mo_types.Async_cap.NullCap with
+    | Error es -> builtin_error "checking" what es
+    | Ok ((_t, sscope), _ws) ->
+      let senv1 = Mo_types.Scope.adjoin senv0 sscope in
+      prog, senv1
+
+let prelude, initial_stat_env0 =
+  check_builtin "prelude" Prelude.prelude Mo_frontend.Typing.initial_scope
+let internals, initial_stat_env =
+  check_builtin "internals" Prelude.internals initial_stat_env0
+
+let show (prog : Mo_def.Syntax.prog Diag.result) : unit =
+  let show_msgs msgs =
+    String.concat "\n" (List.map Diag.string_of_message msgs)
+  in
+  match prog with
+  | Error msgs -> Format.printf "Diagnostics:\n%s\n" (show_msgs msgs)
+  | Ok (prog, msgs) ->
+    let filename = prog.Source.note.Mo_def.Syntax.filename in
+    let module Arrange = Mo_def.Arrange.Make (struct
+      let include_sources = false
+      let include_type_rep = true
+      let include_types = true
+      let include_parenthetical = false
+      let include_docs = None
+      let main_file = Some filename
+    end) in
+    Format.printf "Ok:\n%s\n" (Wasm.Sexpr.to_string 80 (Arrange.prog prog));
+    match msgs with
+    | [] -> ()
+    | _ :: _ -> Format.printf "With diagnostics:\n%s\n" (show_msgs msgs)
+
+let run_test source =
+  Fun.protect
+    (fun () ->
+      Mo_config.Flags.typechecker_combine_srcs := true;
+      let filename = "test-field-srcs.mo" in
+      show begin
+        let open Diag.Syntax in
+        let* prog = parse_from_string Mo_frontend.Lexer.mode filename source in
+        let* prog, _sscope =
+          infer_prog prog initial_stat_env (Mo_types.Async_cap.initial_cap ())
+        in
+        Diag.return prog
+      end)
+    ~finally:(fun () -> Mo_config.Flags.typechecker_combine_srcs := false)
+
+let%expect_test "" =
+  let s = {|actor {
+  class Class1() = self {
+    public func meth(_ : Int) : Nat {
+      return 1
+    }
+  };
+
+  class Class2() = self {
+    public func meth(_ : Nat) : Int {
+      return 2
+    }
+  };
+
+  public func test() : async Int {
+    let c1 : Class2 = Class1();
+    return c1.meth(42)
+  }
+}
+|}
+  in
+  run_test s;
+  [%expect {|
+    Ok:
+    (Prog
+      (ExpD
+        (:
+          (AwaitE
+            (:
+              (AsyncE
+                ($@anon-async-1.1 (: (PrimT Any) Any Any))
+                (:
+                  (ObjBlockE
+                    Actor
```

Comment:
Maybe we can use `src/mo_frontend/tranversals.ml` to gather all `@@ (Pos ..) (Pos ..)` and use that set as the expected value. Ideally, would be gather sources of `meth` only.

Reply:
Or we can just gather map `(symbol, [position])`

Reply:
Done